lib/core/http/fetch-wrapper.ts

```diff
@@ -0,0 +1,58 @@
+import { HttpError } from "@lib/core/http/error";
+
+async function http<T>(path: string, config: RequestInit): Promise<T> {
+  const request = new Request(path, config);
+  const response: Response = await fetch(request);
+
+  if (!response.ok) {
+    const err = HttpError.fromRequest(request, response);
+    throw err;
+  }
+  // may error if there is no body, return empty array
+  return response.json().catch(() => ({}));
```

Comment:
This would break type-safety on some places I'd rather do something like this and just let it throw instead - `({})` suddenly having will be an obscure bug

```ts
 return await response.json();
```